src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -233,876 +233,953 @@ void GetPurchasedAir(EnergyPlusData &state)
-                                                                     NumNums,
-                                                                     IOStat,
-                                                                     s_ipsc->lNumericFieldBlanks,
-                                                                     s_ipsc->lAlphaFieldBlanks,
-                                                                     s_ipsc->cAlphaFieldNames,
-                                                                     s_ipsc->cNumericFieldNames);
-
-            ErrorObjectHeader eoh{routineName, s_ipsc->cCurrentModuleObject, s_ipsc->cAlphaArgs(1)};
-
-            state.dataPurchasedAirMgr->PurchAirNumericFields(PurchAirNum).FieldNames.allocate(NumNums);
-            state.dataPurchasedAirMgr->PurchAirNumericFields(PurchAirNum).FieldNames = "";
-            state.dataPurchasedAirMgr->PurchAirNumericFields(PurchAirNum).FieldNames = s_ipsc->cNumericFieldNames;
-
-            PurchAir(PurchAirNum).Name = s_ipsc->cAlphaArgs(1);
+        auto const &schemaProps = s_ip->getObjectSchemaProps(state, s_ipsc->cCurrentModuleObject);
+        auto &instancesValue = instances_PurchAir.value();
+        for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+
+            ++purchAirNum;
+            auto const &fields = instance.value();
+            std::string thisObjectName = instance.key();
+            auto &PurchAir = state.dataPurchasedAirMgr->PurchAir(purchAirNum);
+            PurchAir.cObjectName = s_ipsc->cCurrentModuleObject;
+            ErrorObjectHeader eoh{routineName, s_ipsc->cCurrentModuleObject, thisObjectName};
+            // Determine max number of alpha and numeric arguments in order to allocate local arrays
+            s_ip->getObjectItem(state,
+                                s_ipsc->cCurrentModuleObject,
+                                purchAirNum,
+                                s_ipsc->cAlphaArgs,
+                                NumAlphas,
+                                s_ipsc->rNumericArgs,
+                                NumNums,
+                                IOStat,
+                                s_ipsc->lNumericFieldBlanks,
+                                s_ipsc->lAlphaFieldBlanks,
+                                s_ipsc->cAlphaFieldNames,
+                                s_ipsc->cNumericFieldNames);
+
+            state.dataPurchasedAirMgr->PurchAirNumericFields(purchAirNum).FieldNames.allocate(NumNums);
+            state.dataPurchasedAirMgr->PurchAirNumericFields(purchAirNum).FieldNames = "";
+            state.dataPurchasedAirMgr->PurchAirNumericFields(purchAirNum).FieldNames = s_ipsc->cNumericFieldNames;
+            Util::IsNameEmpty(state, thisObjectName, s_ipsc->cCurrentModuleObject, ErrorsFound);
+            PurchAir.Name = Util::makeUPPER(thisObjectName);
+            cAlphaFieldName = "Availability Schedule Name";
```

Comment:
I don't think you need this string but am now wondering how CheckUniqueNodeNames (line 312) handles json field name?

Reply:
I considered it during implementation, but I couldn't find a way to do it without saving the field names here. We use these field names in the sizing calculation.  See example below:

`SizingString = state.dataPurchasedAirMgr->PurchAirNumericFields(PurchAirNum).FieldNames(FieldNum) + " [m3/s]";`